pkg/cloudprovider/provider/vsphere/client.go

```diff
@@ -47,6 +50,11 @@ func NewSession(ctx context.Context, config *Config) (*Session, error) {
 		return nil, fmt.Errorf("failed to build client: %v", err)
 	}
 
+	// inject our global set of CA certificates
+	client.DefaultTransport().TLSClientConfig = &tls.Config{
```

Comment:
Check this out what I've found: https://pkg.go.dev/github.com/vmware/govmomi/vim25/soap#Client.SetRootCAs

Reply:
Check out this PR's description:

> The first commit shows the naive approach, where the CA bundle file is given through the stack until it arrives at the cloud provider instances. This was mainly because at the time, I thought the SetRootCAs() function was the only way to inject the CAs in the vSphere provider.

Reply:
well, yeah, but this global state is disgusting :D